### PR TITLE
[Iadvize] Connector evolution and aws-secret provider module

### PR DESCRIPTION
**Pull request by djowalker**
Some evolves for Iadvize-connector.

- Adding more documentation
- Adding a healthcheck for iadvize following the documentation (https://developers.iadvize.com/documentation/build-apps#health-check )
- Add custom data supports
- Add of a module iadvize-client to create and call GraphQL resquests (https://developers.iadvize.com/documentation/graphql-api#overview-graphql)
- Add of aws-tools modules to retrieve aws credentials with a secret provider and add README.md

### CODE REVIEW

File: bot/toolkit-base/src/main/kotlin/BotIoc.kt [hunk 1]
```diff
@@ -37,7 +38,7 @@ object BotIoc {
      * The core modules of the bot.
      */
     val coreModules: List<Module> =
-        listOf(sharedModule, botModule, botMongoModule, noOpTranslatorModule, noOpSTTModule)
+        listOf(sharedModule, botModule, botMongoModule, noOpTranslatorModule, noOpSTTModule, awsToolsModule)
```
Comment (vsct-jburet) [inline bot/toolkit-base/src/main/kotlin/BotIoc.kt:41]: N'ajouter awsToolsModule que si il est dans le classpath

Comment (vsct-jburet) [reply bot/toolkit-base/src/main/kotlin/BotIoc.kt:41]: Use a ServiceLoader:

- define a BotAdditionalModuleService interface in shared
- in aws-tool, add an implementation AwsToolBotAdditionalModuleService of this interface (with a META-INF with file)
- in botIoc, add:

```kotlin
val coreModules: List<Module> =
        listOf(sharedModule, botModule, botMongoModule, noOpTranslatorModule, noOpSTTModule) + loadServices<BotAdditionalModuleService>.map { it.module() }
```

Comment (assouktim) [reply bot/toolkit-base/src/main/kotlin/BotIoc.kt]: All has been done

**Comment by djowalker** [comment]
I tried to move the modules into bot in the other PR, however I encounter somes troubles when launching tests in IntelliJ like :
- Failed to resolve org.junit.platform:junit-platform-launcher:1.10.0.
or
- Failed to resolve org.junit.vintage:junit-vintage-engine:5.10.0.

I'm still looking about that trouble.

**Review by vsct-jburet — APPROVED** [review]

**Pull request merged**